### DISCUSSION TITLE
Slicer Fiducial Registration Wizard

### DISCUSSION THREAD

**Discussion by LeidyMoraV** [PlusToolkit/PlusLib#1188] [Q&A]
Hello!

I've found the Fiducial Registration Wizard module to be helpful for my research. I was wondering if you could provide some documentation to further understand the process behind it? Specifically, the warping transform.

**Comment by lassoan** [comment]
The full source code is available [here](https://github.com/SlicerIGT/SlicerIGT/tree/master/FiducialRegistrationWizard/Logic), but it's your have any specific questions then feel free to ask.

**Reply by LeidyMoraV** [reply]
Thank you! Do you have any publication or related articles to the specific module?

**Reply by lassoan** [reply]
Please cite this paper if you use SlicerIGT:
Ungi T, Lasso A, Fichtinger G. Open-source platforms for navigated image-guided interventions. Med Image Anal. 2016 Oct;33:181-6. doi: 10.1016/j.media.2016.06.011.

If you need citation on any specific algorithm then we can help you looking it up (the references are in the source code).

**Reply by LeidyMoraV** [reply]
I'm looking for a document that explains the mathematical procedures behind Fiducial Registration Wizard module

**Comment by adamrankin** [comment]
From the link that Andras gave you we read the code here:
https://github.com/SlicerIGT/SlicerIGT/blob/b54955f2e57ba9b673f9635cae82e6cf8b91eb36/FiducialRegistrationWizard/Logic/vtkSlicerFiducialRegistrationWizardLogic.cxx#L361

Googling "vtkLandmarkTransform" sent me here
https://vtk.org/doc/nightly/html/classvtkLandmarkTransform.html#details
which didn't explain the algorithm so I google'd the source here
https://github.com/Kitware/VTK/blob/master/Common/Transforms/vtkLandmarkTransform.cxx

Reading the code led me to this:
https://github.com/Kitware/VTK/blob/1b05133f4431793378074958a9880880dcc64779/Common/Transforms/vtkLandmarkTransform.cxx#L70

In the future, it would be a valuable exercise in research skills to try to find these things on your own

**Reply by adamrankin** [reply]
```
/*
    The solution is based on
    Berthold K. P. Horn (1987),
    "Closed-form solution of absolute orientation using unit quaternions,"
    Journal of the Optical Society of America A, 4:629-642
  */
```

**Reply by ungi** [reply]
Hi @adamrankin :) I think the original question mentioned warping transform. The default for warping method is thin plate spines. VTK says it is from here: https://ieeexplore.ieee.org/document/24792
